src/gui/modalMenu.cpp

```diff
@@ -230,89 +288,50 @@ bool GUIModalMenu::preprocessEvent(const SEvent &event)
 	}
 
 	if (event.EventType == EET_TOUCH_INPUT_EVENT) {
-		SEvent translated;
-		memset(&translated, 0, sizeof(SEvent));
-		translated.EventType = EET_MOUSE_INPUT_EVENT;
-		gui::IGUIElement *root = Environment->getRootGUIElement();
-
-		if (!root) {
-			errorstream << "GUIModalMenu::preprocessEvent"
-				    << " unable to get root element" << std::endl;
-			return false;
-		}
-		gui::IGUIElement *hovered =
-				root->getElementFromPoint(core::position2d<s32>(
-						event.TouchInput.X, event.TouchInput.Y));
-
-		translated.MouseInput.X = event.TouchInput.X;
-		translated.MouseInput.Y = event.TouchInput.Y;
-		translated.MouseInput.Control = false;
+		irr_ptr<GUIModalMenu> holder;
+		holder.grab(this); // keep this alive until return (it might be dropped downstream [?])
```

Comment:
Why not use the `explicit irr_ptr(ReferenceCounted *object) noexcept` constructor here?

Reply:
Because it has move semantics. It would be nicer to initialize it with `grab` from d0222e5c421b6580ccc4c3bdee1c080a48f9b7e2 (from #10658) instead, though.